rules/typescript.js

```diff
@@ -0,0 +1,68 @@
+// eslint-disable-next-line import/no-commonjs
+module.exports = {
+  plugins: ['@typescript-eslint'],
+  rules: {
+    '@typescript-eslint/adjacent-overload-signatures': ['error'],
+    '@typescript-eslint/array-type': ['error', 'array-simple'],
+    '@typescript-eslint/ban-types': ['error'],
+    '@typescript-eslint/ban-ts-ignore': ['off'],
+    '@typescript-eslint/camelcase': ['error'],
+    '@typescript-eslint/class-name-casing': ['error'],
+    '@typescript-eslint/explicit-function-return-type': ['off'],
+    '@typescript-eslint/explicit-member-accessibility': ['error'],
+    '@typescript-eslint/generic-type-naming': ['off'],
```

Comment:
Would be nice to enable it to force the usage of `TXxxx` inside a generic.

Reply:
No objection on this. There will be 10 errors to fix in InstantSearch.js.